src/sql/workbench/services/query/common/queryManagement.ts

```diff
@@ -339,16 +339,15 @@ export class QueryManagementService implements IQueryManagementService {
 		let item = this._queryRunners.get(oldUri);
 		if (!item) {
 			this._logService.error(`No query runner found for old URI : '${oldUri}'`);
-			throw new Error(nls.localize('queryManagement.noQueryRunnerForUri', 'Could not find Query Runner for uri: {0}', oldUri));
-		}
-		if (this._queryRunners.get(newUri)) {
+		} else if (this._queryRunners.get(newUri)) {
 			this._logService.error(`New URI : '${newUri}' already has a query runner.`);
```

Comment:
We probably shouldn't be changing the URI in this case since that could cause incorrect results for anything else using that URI. Better to fail out immediately IMO. Ideally we'd also want to clean up/disconnect the old query URI too, although that may need to be a follow up item to look in to if it ends up being a huge mess...

Reply:
ok, added the throw new Error part back in.